pandas/tests/series/test_operators.py

```diff
@@ -28,6 +28,31 @@
 from .common import TestData
 
 
+class TestDatetimeLikeArithmetic(object):
+    def test_sub_datetime64_not_ns(self):
+        # GH#7996
+        ser = Series(date_range('20130101', periods=3))
+        dt64 = np.datetime64('2013-01-01')
+        assert dt64.dtype == 'datetime64[D]'
+        res = ser - dt64
+        expected = pd.Series([Timedelta(days=0), Timedelta(days=1),
+                              Timedelta(days=2)])
+        tm.assert_series_equal(res, expected)
+
+        res = dt64 - ser
+        tm.assert_series_equal(res, -expected)
+
+        # check for DatetimeIndex and DataFrame while we're at it
+        dti = pd.DatetimeIndex(ser)
+        res = dti - dt64
+        tm.assert_index_equal(res, pd.Index(expected))
+
+        res = dt64 - dti
+        tm.assert_index_equal(res, pd.Index(-expected))
+
+        # TODO: This is still broken for ser.to_frame()
```

Comment:
then if you want to add a test with an xfail would be helpful. (and an issue reference), TODO's are only so good.